### PR TITLE
Expand/align promise/rx tx examples

### PR DESCRIPTION
**Pull request by jacogr**

### CODE REVIEW

**Comment by ltfschoen** [comment]
@jacogr I tried running `yarn run test` after removing the `skip` from the tests. And whilst running the latest substrate node in the "master" branch and updating it with these instructions https://github.com/paritytech/substrate#61-hacking-on-substrate. I also tried using the "v0.2" branch but got the same error

But I get the following error:
```
 FAIL  packages/api/test/e2e/promise-tx.spec.js
  ● e2e transactions › makes a transfer

    ExtError: Unable to determine type from '::Type'

      151 |   const Type = Types[value.type];
      152 | 
    > 153 |   assert(Type, `Unable to determine type from '${value.type}'`);
          |   ^
      154 | 
      155 |   return Type;
      156 | }

      at assert (node_modules/@polkadot/util/assert.js:44:9)
      at assert (packages/types/src/codec/createType.ts:153:3)
      at getTypeClass (packages/types/src/Method.ts:150:20)
          at Array.reduce (<anonymous>)
      at Function.reduce [as getArgsDef] (packages/types/src/Method.ts:149:38)
      at Function.getArgsDef [as decodeMethod] (packages/types/src/Method.ts:106:30)
      at new decodeMethod (packages/types/src/Method.ts:50:28)
      at extrinsicFn (packages/type-extrinsics/src/utils/createUnchecked.ts:33:15)
      at Object.method [as transfer] (packages/api/src/promise/index.ts:209:38)
      at Object.transfer (packages/api/test/e2e/promise-tx.spec.js:28:8)

  ● e2e transactions › makes a proposal

    ExtError: Unable to determine type from '::Type'

      151 |   const Type = Types[value.type];
      152 | 
    > 153 |   assert(Type, `Unable to determine type from '${value.type}'`);
          |   ^
      154 | 
      155 |   return Type;
      156 | }

      at assert (node_modules/@polkadot/util/assert.js:44:9)
      at assert (packages/types/src/codec/createType.ts:153:3)
      at getTypeClass (packages/types/src/Method.ts:150:20)
          at Array.reduce (<anonymous>)
      at Function.reduce [as getArgsDef] (packages/types/src/Method.ts:149:38)
      at Function.getArgsDef [as decodeMethod] (packages/types/src/Method.ts:106:30)
      at new decodeMethod (packages/types/src/Method.ts:50:28)
      at extrinsicFn (packages/type-extrinsics/src/utils/createUnchecked.ts:33:15)
      at Object.method [as propose] (packages/api/src/promise/index.ts:209:38)
      at Object.propose (packages/api/test/e2e/promise-tx.spec.js:44:8)


Test Suites: 1 failed, 10 skipped, 71 passed, 72 of 82 total
Tests:       2 failed, 23 skipped, 298 passed, 323 total
Snapshots:   0 total
Time:        13.151s, estimated 14s
Ran all test suites.

```

**Comment by jacogr** [comment]
It means `::Type` is invalid and not recognised, nothing to do with transfers. The metadata parsing now goes wrong - as it stands master won't yet work, since there is the Compact encoding in now for some fields (we have the encoders, but have not checked those) and 0.2 won't work either since that is not "latest API compliant", i.e. no metadata there.

Since master has not been accessible while SHAFT was WIP and SHAFT has just been merged - even worse, BBQ doesn't support Compact BlockNumber yet so adding support breaks that one, for which there is no branch anymore...

TL;DR Issues is caused by not having support for latest meta yet - and this is an issue, unless BBQ jumps it is a tightrope.
